spot-ml/src/main/scala/org/apache/spot/SpotLDAWrapper.scala

```diff
@@ -0,0 +1,189 @@
+package org.apache.spot
+
+import org.apache.log4j.Logger
+import org.apache.spark.SparkContext
+import org.apache.spark.mllib.clustering.{DistributedLDAModel, EMLDAOptimizer, LDA}
+import org.apache.spark.mllib.linalg.{Matrix, Vector, Vectors}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+import org.apache.spot.SpotLDAWrapperSchema._
+
+import scala.collection.immutable.Map
+
+/**
+  * Spark LDA implementation
+  * Contains routines for LDA using Scala Spark implementation from mllib
+  * 1. Creates list of unique documents, words and model based on those two
+  * 2. Processes the model using Spark LDA
+  * 3. Reads Spark LDA results: Topic distributions per document (docTopicDist) and word distributions per topic (wordTopicMat)
+  * 4. Convert wordTopicMat (Matrix) and docTopicDist (RDD) to appropriate formats (maps) originally specified in LDA-C
+  */
+
+object SpotLDAWrapper {
+
+  case class SpotLDAInput(doc: String, word: String, count: Int) extends Serializable
+
+  case class SpotLDAOutput(docToTopicMix: DataFrame, wordResults: Map[String, Array[Double]])
+
+
+  def runLDA(sparkContext: SparkContext,
+             sqlContext: SQLContext,
+             docWordCount: RDD[SpotLDAInput],
+             topicCount: Int,
+             logger: Logger,
+             ldaSeed: Option[Long],
+             ldaAlpha: Double = 1.02,
+             ldaBeta: Double = 1.001,
+             maxIterations: Int = 20): SpotLDAOutput = {
```

Comment:
You mean to leave all defaults as parameters? We can do that, we would just need to explain to users what values to pass and what we have tested. Is that what you are saying?